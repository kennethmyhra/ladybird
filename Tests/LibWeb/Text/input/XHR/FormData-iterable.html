<script src="../include.js"></script>
<script>
    test(() => {
        const formData = new FormData();
        formData.append("key1", "value1");
        formData.append("key2", "value2");
        formData.append("key2", new File([], "file_name.txt"));

        println("Iterate keys in a for loop");
        for (const key of formData.keys())
            println(key);

        println("\nIterate values in a for loop");
        for (const value of formData.values())
            println(value);

        println("\nIterate entries in a for loop");
        for (const pair of formData.entries())
            println(`${pair[0]}, ${pair[1]}`);

        println("\nIterate values by explicitly calling IterableIterator.next()");
        const iterator = formData.values();
        for (; ;) {
            let result = iterator.next();
            if (result.done)
                break;

            const value = result.value;
            println(value);
        }

        println("\nIterate values by calling .forEach()");
        formData.forEach((value, key, _) => {
            println(`${value}, ${key}`);
        });

    });
</script>
