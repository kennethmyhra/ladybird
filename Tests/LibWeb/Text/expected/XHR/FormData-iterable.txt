Iterate keys in a for loop
key1
key2
key2

Iterate values in a for loop
value1
value2
[object File]

Iterate entries in a for loop
key1, value1
key2, value2
key2, [object File]

Iterate values by explicitly calling IterableIterator.next()
value1
value2
[object File]

Iterate values by calling .forEach()
value1, key1
value2, key2
[object File], key2
